carmm/run/workflows/react.py

```diff
@@ -579,32 +580,36 @@ def _calc_generator(params,
     from carmm.run.aims_calculator import get_aims_and_sockets_calculator
     """On machines where ASE and FHI-aims are run separately (e.g. ASE on login node, FHI-aims on compute nodes)
     we need to specifically state what the name of the login node is so the two packages can communicate"""
-    sockets_calc, fhi_calc = get_aims_and_sockets_calculator(dimensions=dimensions,
-                                                             logfile=f"{directory}/socketio.log",
-                                                             verbose=True,
-                                                             codata_warning=False,
-                                                             directory=directory
-                                                             )
 
     """Remove previous xc argument to ensure libxc warning override is first"""
-    fhi_calc.parameters = {k: v for k, v in fhi_calc.parameters.items() if k != 'xc'}
-    fhi_calc.parameters['override_warning_libxc'] ='True'
+    parameters = dict(params)
+    xc = parameters['xc']
+    del parameters['xc']
+    if "libxc" in xc:
+        parameters['override_warning_libxc'] ='True'
+    parameters['xc'] = xc
 
     """Forces required for optimisation"""
-    if not forces:
-        fhi_calc.parameters = {k: v for k, v in fhi_calc.parameters.items() if k != 'compute_forces'}
+    if not forces and ('compute_forces' in parameters):
+        del parameters['compute_forces']
```

Comment:
Same comment as above, these seem identical. Perhaps needs @ikowalec to review.

Reply:
I changed the `if` statement here because it will throw an error when deleting `parameters['compute_forces']` if `compute_forces` is not set in parameters.

Reply:
Which if? What is the error? The conditional statement in dictionary comprehension will not throw an error if the string does not exist in the dictionary.

Reply:
line 593.

It does give an error. Here is an example.

> mydict = {'arm':2}
> del mydict['feet']
> Traceback (most recent call last):
>   File "/usr/lib64/python3.11/code.py", line 90, in runcode
>     exec(code, self.locals)
>   File "<input>", line 1, in <module>
> KeyError: 'feet'
![image](https://github.com/user-attachments/assets/0336b799-ca97-45d4-998d-f71712d91b1b)

Reply:
I think the point from prior is that the delete isn't needed with Igor's command - it instead only copies the relevant information in the first place, and doesn't copy in `compute_force`.

e.g. `mydict = {k: v for k, v in ['arm':2] if k != 'feet'}`

I'd say it's cleaner to copy in just what is needed, rather than copy everything in and then delete items?

Reply:
Resolved as the original code is retained.